src/main/scala/is/hail/rvd/RVDContext.scala

```diff
@@ -1,20 +1,58 @@
 package is.hail.rvd
 
-import is.hail.annotations.Region
+import is.hail.annotations.{Region, RegionValueBuilder}
+import is.hail.sparkextras.ResettableContext
+
+import scala.collection.mutable
 
 object RVDContext {
   def default: RVDContext = new RVDContext(Region())
 
   def fromRegion(region: Region): RVDContext = new RVDContext(region)
 }
 
-// NB: must be *Auto*Closeable because calling close twice is undefined behavior
-// (see AutoCloseable javadoc)
-class RVDContext(r: Region) extends AutoCloseable {
+class RVDContext(r: Region) extends ResettableContext {
+  private[this] val children: mutable.ArrayBuffer[AutoCloseable] = new mutable.ArrayBuffer()
+
+  private[this] def own(child: AutoCloseable): Unit = children += child
+
+  own(r)
+
+  def freshContext: RVDContext = {
+    val ctx = RVDContext.default
+    own(ctx)
+    ctx
+  }
+
   def region: Region = r // lifetime: element
```

Comment:
Remark: I think the lifetime is really up to the caller.  I think of this "the place where to allocate memory to return my value".